From: John Rigby <jrigby@freescale.com>
Date: Thu, 24 Aug 2006 15:13:35 -0600
Subject: [PATCH] mxc platform

copied custom-linux-mxc-g++.* from qtopia-free-2.1.1 mxc patch
which in turn are modified versions of custom-linux-ipaq-g++.*
in the qtopia free distribution.

Signed-off-by: John Rigby <jrigby@freescale.com>

---

 .../src/libraries/qtopia/custom-linux-mxc-g++.cpp  |   92 +++++++++++++++++++++++
 qtopia/src/libraries/qtopia/custom-linux-mxc-g++.h |   46 ++++++++++
 2 files changed, 138 insertions(+), 0 deletions(-)
 create mode 100644 qtopia/src/libraries/qtopia/custom-linux-mxc-g++.cpp
 create mode 100644 qtopia/src/libraries/qtopia/custom-linux-mxc-g++.h

0d574598feb2d45e9c557cbedbd23eb9e8a22aad
diff --git a/qtopia/src/libraries/qtopia/custom-linux-mxc-g++.cpp b/qtopia/src/libraries/qtopia/custom-linux-mxc-g++.cpp
new file mode 100644
index 0000000..80e7d63
--- /dev/null
+++ b/qtopia/src/libraries/qtopia/custom-linux-mxc-g++.cpp
@@ -0,0 +1,92 @@
+/**********************************************************************
+** Copyright (C) 2000-2005 Trolltech AS.  All rights reserved.
+**
+** This file is part of the Qtopia Environment.
+** 
+** This program is free software; you can redistribute it and/or modify it
+** under the terms of the GNU General Public License as published by the
+** Free Software Foundation; either version 2 of the License, or (at your
+** option) any later version.
+** 
+** A copy of the GNU GPL license version 2 is included in this package as 
+** LICENSE.GPL.
+**
+** This program is distributed in the hope that it will be useful, but
+** WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+** See the GNU General Public License for more details.
+**
+** In addition, as a special exception Trolltech gives permission to link
+** the code of this program with Qtopia applications copyrighted, developed
+** and distributed by Trolltech under the terms of the Qtopia Personal Use
+** License Agreement. You must comply with the GNU General Public License
+** in all respects for all of the code used other than the applications
+** licensed under the Qtopia Personal Use License Agreement. If you modify
+** this file, you may extend this exception to your version of the file,
+** but you are not obligated to do so. If you do not wish to do so, delete
+** this exception statement from your version.
+** 
+** See http://www.trolltech.com/gpl/ for GPL licensing information.
+**
+** Contact info@trolltech.com if any conditions of this licensing are
+** not clear to you.
+**
+**********************************************************************/
+
+#include "qtopia/power.h"
+
+#include <qwindowsystem_qws.h>
+#include <linux/fb.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <qfile.h>
+
+int qpe_sysBrightnessSteps()
+{
+    return 255;
+}
+
+
+void qpe_setBrightness(int bright)
+{
+    if ( QFile::exists("/usr/bin/bl") ) {
+	QString cmd = "/usr/bin/bl 1 ";
+	cmd += bright<=0 ? "0 " : "1 ";
+	cmd += QString::number(bright);
+	system(cmd.latin1());
+    } 
+}
+
+void PowerStatusManager::getStatus()
+{
+}
+
+typedef struct KeyOverride {
+    ushort scan_code;
+    QWSServer::KeyMap map;
+};
+
+static const KeyOverride deviceKeys[] = {
+   { 0x79,   {   Qt::Key_F34,    0xffff, 0xffff, 0xffff  } }, // power
+#if defined(QT_KEYPAD_MODE)
+   { 0x7a,   {   Qt::Key_Call,	    0xffff, 0xffff, 0xffff  } },
+   { 0x7b,   {   Qt::Key_Context1,  0xffff, 0xffff, 0xffff  } },
+   { 0x7c,   {   Qt::Key_Back,	    0xffff, 0xffff, 0xffff  } },
+   { 0x7d,   {   Qt::Key_Hangup,    0xffff, 0xffff, 0xffff  } },
+#else
+   { 0x7a,   {   Qt::Key_F9,	    0xffff, 0xffff, 0xffff  } },
+   { 0x7b,   {   Qt::Key_F10,	    0xffff, 0xffff, 0xffff  } },
+   { 0x7c,   {   Qt::Key_F11,	    0xffff, 0xffff, 0xffff  } },
+   { 0x7d,   {   Qt::Key_F12,	    0xffff, 0xffff, 0xffff  } },
+#endif
+   { 0,	     {   0,		    0xffff, 0xffff, 0xffff  } }
+};
+
+const KeyOverride* qtopia_override_keys()
+{
+    return deviceKeys;
+}
diff --git a/qtopia/src/libraries/qtopia/custom-linux-mxc-g++.h b/qtopia/src/libraries/qtopia/custom-linux-mxc-g++.h
new file mode 100644
index 0000000..522622e
--- /dev/null
+++ b/qtopia/src/libraries/qtopia/custom-linux-mxc-g++.h
@@ -0,0 +1,46 @@
+/**********************************************************************
+** Copyright (C) 2000-2005 Trolltech AS.  All rights reserved.
+**
+** This file is part of the Qtopia Environment.
+** 
+** This program is free software; you can redistribute it and/or modify it
+** under the terms of the GNU General Public License as published by the
+** Free Software Foundation; either version 2 of the License, or (at your
+** option) any later version.
+** 
+** A copy of the GNU GPL license version 2 is included in this package as 
+** LICENSE.GPL.
+**
+** This program is distributed in the hope that it will be useful, but
+** WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+** See the GNU General Public License for more details.
+**
+** In addition, as a special exception Trolltech gives permission to link
+** the code of this program with Qtopia applications copyrighted, developed
+** and distributed by Trolltech under the terms of the Qtopia Personal Use
+** License Agreement. You must comply with the GNU General Public License
+** in all respects for all of the code used other than the applications
+** licensed under the Qtopia Personal Use License Agreement. If you modify
+** this file, you may extend this exception to your version of the file,
+** but you are not obligated to do so. If you do not wish to do so, delete
+** this exception statement from your version.
+** 
+** See http://www.trolltech.com/gpl/ for GPL licensing information.
+**
+** Contact info@trolltech.com if any conditions of this licensing are
+** not clear to you.
+**
+**********************************************************************/
+
+#define QPE_NEED_CALIBRATION
+#define QPE_USE_MALLOC_FOR_NEW
+#define QPE_ARCHITECTURE "Freescale/MXC"
+#define QPE_DEFAULT_TODAY_MODE "Daily"
+
+#define QPE_OVERRIDE_KEYMAP
+struct KeyOverride;
+extern const KeyOverride* qtopia_override_keys();
+
+
+//#define QPE_OVERRIDE_KEYHANDLER
-- 
1.3.3

